selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/GolangFileEditingTest.java

```diff
@@ -0,0 +1,158 @@
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.selenium.languageserver;
+
+import static org.eclipse.che.selenium.core.constant.TestMenuCommandsConstants.Assistant.ASSISTANT;
+import static org.eclipse.che.selenium.core.constant.TestMenuCommandsConstants.Assistant.FIND_DEFINITION;
+import static org.eclipse.che.selenium.core.workspace.WorkspaceTemplate.UBUNTU_GO;
+import static org.eclipse.che.selenium.pageobject.CodenvyEditor.ContextMenuLocator.FORMAT;
+import static org.eclipse.che.selenium.pageobject.CodenvyEditor.MarkerLocator.ERROR;
+import static org.openqa.selenium.Keys.F4;
+
+import com.google.inject.Inject;
+import java.net.URL;
+import java.nio.file.Paths;
+import org.eclipse.che.selenium.core.client.TestProjectServiceClient;
+import org.eclipse.che.selenium.core.project.ProjectTemplates;
+import org.eclipse.che.selenium.core.workspace.InjectTestWorkspace;
+import org.eclipse.che.selenium.core.workspace.TestWorkspace;
+import org.eclipse.che.selenium.pageobject.CodenvyEditor;
+import org.eclipse.che.selenium.pageobject.Consoles;
+import org.eclipse.che.selenium.pageobject.Ide;
+import org.eclipse.che.selenium.pageobject.Menu;
+import org.eclipse.che.selenium.pageobject.ProjectExplorer;
+import org.openqa.selenium.Keys;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/** @author Skoryk Serhii */
+public class GolangFileEditingTest {
+
+  private static final String PROJECT_NAME = "desktop-go-simple";
+  private static final String GO_FILE_NAME = "main.go";
+  private static final String PATH_TO_GO_FILE = PROJECT_NAME + "/" + GO_FILE_NAME;
+  private static final String LS_INIT_MESSAGE = "Finished running tool: /usr/local/go/bin/go build";
+  private static final String FORMATTED_CODE =
+      "package main\n"
+          + "\n"
+          + "import (\n"
+          + " \"fmt\"\n"
+          + " \"math\"\n"
+          + ")\n"
+          + "\n"
+          + "func print() {\n"
+          + " fmt.Printf(\"Hello, world. Sqrt(2) = %v\\n\", math.Sqrt(2))\n"
+          + "}";
+
+  @InjectTestWorkspace(template = UBUNTU_GO)
+  private TestWorkspace workspace;
+
+  @Inject private Ide ide;
+  @Inject private Menu menu;
+  @Inject private Consoles consoles;
+  @Inject private CodenvyEditor editor;
+  @Inject private ProjectExplorer projectExplorer;
+  @Inject private TestProjectServiceClient testProjectServiceClient;
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    URL resource = ApacheCamelFileEditingTest.class.getResource("/projects/go-simple");
+    testProjectServiceClient.importProject(
+        workspace.getId(), Paths.get(resource.toURI()), PROJECT_NAME, ProjectTemplates.GO);
+
+    ide.open(workspace);
+  }
+
+  @Test
+  public void checkLanguageServerInitialized() {
+    projectExplorer.waitAndSelectItem(PROJECT_NAME);
```

Comment:
```
    projectExplorer.waitAndSelectItem(PROJECT_NAME);
    projectExplorer.openItemByPath(PROJECT_NAME);
    projectExplorer.openItemByPath(PATH_TO_GO_FILE);
```
Can be changed to "ProjectExplorer#expandPathInProjectExplorerAndOpenFile(String path, String fileName)"

Reply:
Can we apply expandPathInProjectExplorerAndOpenFile?For reducing of opening time.